modules/siddhi-extensions/input-mappers/json-input-mapper/src/main/java/org/wso2/siddhi/extension/input/mapper/json/JsonSourceMapper.java

```diff
@@ -0,0 +1,515 @@
+                        log.error(e.getMessage());
+                    }
+                }
+                Event[] eventArray = new Event[eventList.size()];
+                eventArray = eventList.toArray(eventArray);
+                return eventArray;
+            } else {
+                try {
+                    Event event = processCustomEvent(JsonPath.parse(jsonObj));
+                    return event;
+                } catch (ExecutionPlanRuntimeException e) {
+                    log.error(e.getMessage());
+                    return null;
+                }
+            }
+        } else {
+            jsonObj = readContext.read(DEFAULT_ENCLOSING_ELEMENT);
+            if (jsonObj instanceof JSONArray) {
+                return convertToEventArrayForDefaultMapping(eventObject);
+            } else {
+                try {
+                    Event event = convertToSingleEventForDefaultMapping(eventObject);
+                    if (event == null) {
+                        log.error("Event" + eventObject + " contains incompatible attribute types and values.");
+                        return null;
+                    }
+                    return event;
+                } catch (IOException e) {
+                    log.error("Invalid JSON string :" + eventObject.toString() + ". Cannot be converted to an event.");
+                    return null;
+                } catch (ExecutionPlanRuntimeException e) {
+                    log.error(e.getMessage());
+                    return null;
+                }
+            }
+        }
+    }
+
+    private Event convertToSingleEventForDefaultMapping(Object eventObject) throws IOException {
+        Event event = new Event(attributesSize);
+        Object[] data = event.getData();
+        com.fasterxml.jackson.core.JsonParser parser;
+        int numberOfProvidedAttributes = 0;
+        try {
+            parser = factory.createParser(eventObject.toString());
+        } catch (IOException e) {
+            throw new ExecutionPlanRuntimeException("Initializing a parser failed for the event string."
+                    + eventObject.toString());
+        }
+        int position;
+        while (!parser.isClosed()) {
+            JsonToken jsonToken = parser.nextToken();
+            if (JsonToken.START_OBJECT.equals(jsonToken)) {
+                parser.nextToken();
+                if (DEFAULT_JSON_EVENT_IDENTIFIER.equalsIgnoreCase(parser.getText())) {
+                    parser.nextToken();
+                } else {
+                    throw new ExecutionPlanRuntimeException("Default json event " + eventObject
+                            + " contains an invalid event identifier. Required \"event\", " +
+                            "but found \"" + parser.getText() + "\".");
+                }
+            } else if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+                String key = parser.getCurrentName();
+                numberOfProvidedAttributes++;
```

Comment:
Since the json validation is done before this method call, the input string will be a valid json object. Therefore it will iterate through the object until the parser finds an start element of a json string or a Field name. Then it will do the process accordingly.